scripts/trento-support.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+set -e
+
+readonly ARGS=("$@")
+declare -A VALID_FACILITIES=(
+    ["configuration"]="configuration"
+    ["base"]="base"
+    ["kubernetes"]="kubernetes"
+    ["all"]="all"
+)
+
+indent() { sed 's/^/  /'; }
+
+collect_trento_configuration() {
+    echo "#==[ Configuration File ]===========================#"
+    echo "# /etc/trento/installer.conf"
+    echo "$(</etc/trento/installer.conf)"    
+} &> "$OUTPUT"
+
+collect_base_system() {
+    echo "#==[ Command ]======================================#"
+    echo "# k3s --version"
+    k3s --version
+
+    echo "#==[ Command ]======================================#"
+    echo "# $(which helm) version"
+    helm version
+
+    echo "#==[ Command ]======================================#"
+    echo "# $(which helm) get hooks trento-server"
+    helm get hooks trento-server
+    
+    echo "#==[ Command ]======================================#"
+    echo "# $(which helm) get manifest trento-server"
+    helm get manifest trento-server | yq -n '[inputs]' | jq 'walk(if type == "object" then del(.data.privatekey, .data."postgresql-password", .data."postgresql-postgres-password", .secretKeyRef, ."admin-user", ."admin-password", ."SMTP_PASSWORD", ."ADMIN_USER", ."ADMIN_PASSWORD", ."SECRET_KEY_BASE") else . end)'
+    
+    echo "#==[ Command ]======================================#"
+    echo "# $(which helm) get notes trento-server"
+    helm get notes trento-server
+    
+    echo "#==[ Command ]======================================#"
+    echo "# $(which helm) get values trento-server"
+    helm get values trento-server | yq 'del(."trento-runner".privateKey, ."trento-web".adminUser)'
+} &> "$OUTPUT"
+
+collect_kubernetes_state() {
+    echo "#==[ Command ]======================================#"
+    echo "# $(which kubectl) get nodes -o wide"
+    kubectl get nodes -o wide
+
+    echo "#==[ Command ]======================================#"
+    echo "# $(which kubectl) get pods"
+    kubectl get pods
+
+    echo "#==[ Command ]======================================#"
+    echo "# $(which kubectl) logs deploy/trento-server-runner"
+    kubectl logs deploy/trento-server-runner
+
+    echo "#==[ Command ]======================================#"
+    echo "# $(which kubectl) logs deploy/trento-server-web -c init"
+    kubectl logs deploy/trento-server-web -c init
+
+    echo "#==[ Command ]======================================#"
+    echo "# $(which kubectl) logs deploy/trento-server-web"
+    kubectl logs deploy/trento-server-web
+
+    echo "#==[ Command ]======================================#"
+    echo "# $(which kubectl) describe deployments" 
+    kubectl describe deployments
+
+    echo "#==[ Command ]======================================#"
+    echo "# $(which crictl) images"
+    crictl images
```

Comment:
It's true that both `k3s` and `crictl` might not be present for someone who installed helm on his own and used the helm charts without the script. Adding a dependency on the spec file in this case might be a bad idea as long as we have the install script around since it's going to install conflicting binaries.

For now, what makes the most sense to me is to check for the `crictl` binary and `k3s` binary, attempt to use if available and otherwise skip the parts where we use it, issuing maybe a warning that the binary wasn't there. For the `helm` binary missing case I'd directly abort the execution.

Reply:
> For now, what makes the most sense to me is to check for the `crictl` binary and `k3s` binary, attempt to use if available and otherwise skip the parts where we use it, issuing maybe a warning that the binary wasn't there. For the `helm` binary missing case I'd directly abort the execution.

sounds good!